### ISSUE TITLE
Commands

### ISSUE THREAD

**Issue by Hendrik-Sama** [SteamingMutt/WildBeast#1]
Instead of ~catfacts and ~facts, you could have  ~Facts or ~Facts Cat so you could later import other kinds of facts but you don't have to create a completely new command for them.

**Comment by Perpetucake** [comment]
The !facts command honestly was mostly implemented as a proof of concept for testing xml2js. The datasource is shit and I agree that it needs a rewrite, including this feature. The easiest way to do this is find a datasource of a json file with facts organized by category. Then it'd be as simple as grabbing fact.category[RNG], or without suffix it'd just do fact.RNG[RNG]. Pretty easy to write. An issue with this of course is the same as with the (unfinished) !trivia function. Shipping around a GIANT json file filled with facts (or trivia) is unrealistic (53MB file can't even be opened in a text editor on my PC lol). This is why for !trivia an api based on that file will be used http://jservice.io/. And honestly we could use that as a !fact function with categories if we really wanted to, as it includes the ability to grab random trivia questions OR grab them by categories. Decentralizing the fact source also means that it is harder to update and will get really annoying if the bot is forked etc etc
Cat facts is a dank meme though, and will stay forever separate. It may become a subscription service that requires an admin to unsubscribe you though, to follow in the history of the catfacts meme.

tl;dr need a better facts api, if it had categories this would already be implemented. probably will just say fuck facts and do trivia with point tracking (when I get the motivation), cat facts is too dank of a meme to include in a facts command. If facts ends up staying in for the long haul i'll make !facts cat work just for you though, but I believe trivia will replace it as it is a more versatile and valuable function not based off of a shitty api.

tl;dr tl;dr trivia will replace facts and catfacts is a dank maymay which will stand alone forever
